gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1492,6 +1510,9 @@ private String listStorageObjectsAndPrefixesPage(
         logger.atFiner().withCause(e).log(
             "listStorageObjectsAndPrefixesPage(%s, %s): item not found", resource, listOptions);
         return null;
+      } else if (ApiErrorExtractor.INSTANCE.requestFailure(e)) {
+        gcsStatistics.putIfAbsent(ACTION_HTTP_GET_REQUEST_FAILURES, new AtomicLong(1));
+        gcsStatistics.get(ACTION_HTTP_GET_REQUEST_FAILURES).incrementAndGet();
```

Comment:
Instead of capturing exceptions every time, please use response interceptor `HttpResponseInterceptor`

Reply:
bump